tests/test_perf_analyzer.py

```diff
@@ -226,110 +229,162 @@ def test_run(self):
         self.server.start()
         self.client.wait_for_server_ready(num_retries=1)
 
-        # Run perf analyzer with dummy metrics to check command parsing
-        perf_metrics = [id]
-        test_latency_output = "Client:\n  p99 latency: 5000 us\n\n\n\n"
-        self.perf_mock.set_perf_analyzer_result_string(test_latency_output)
-        with self.assertRaises(TritonModelAnalyzerException):
-            perf_analyzer.run(perf_metrics)
+        pa_csv_mock = """Concurrency,Inferences/Second,Client Send,Network+Server Send/Recv,Server Queue,Server Compute Input,Server Compute Infer,Server Compute Output,Client Recv,p50 latency,p90 latency,p95 latency,p99 latency,Avg latency,request/response,response wait\n1,46.8,2,187,18,34,65,16,1,4600,4700,4800,4900,5000,3,314"""
 
         # Test avg latency parsing
-        test_latency_output = "Client:\n  Avg latency: 5000 us\n\n\n\n"
-        self.perf_mock.set_perf_analyzer_result_string(test_latency_output)
         perf_metrics = [PerfLatencyAvg]
-        perf_analyzer.run(perf_metrics)
+
+        with patch('model_analyzer.perf_analyzer.perf_analyzer.open',
+                   mock_open(read_data=pa_csv_mock)), patch(
+                       'model_analyzer.perf_analyzer.perf_analyzer.os.remove'):
+            perf_analyzer.run(perf_metrics)
+
         records = perf_analyzer.get_records()
         self.assertEqual(len(records), 1)
         self.assertEqual(records[0].value(), 5)
 
         # Test p90 latency parsing
-        test_latency_output = "Client:\n  p90 latency: 5000 us\n\n\n\n"
-        self.perf_mock.set_perf_analyzer_result_string(test_latency_output)
         perf_metrics = [PerfLatencyP90]
-        perf_analyzer.run(perf_metrics)
+
+        with patch('model_analyzer.perf_analyzer.perf_analyzer.open',
+                   mock_open(read_data=pa_csv_mock)), patch(
+                       'model_analyzer.perf_analyzer.perf_analyzer.os.remove'):
+            perf_analyzer.run(perf_metrics)
+
         records = perf_analyzer.get_records()
         self.assertEqual(len(records), 1)
-        self.assertEqual(records[0].value(), 5)
+        self.assertEqual(records[0].value(), 4.7)
```

Comment:
did the test data change? or was a rounding bug fixed?

Reply:
I changed the test data